nautobot_circuit_maintenance/metrics_app.py

```diff
@@ -0,0 +1,81 @@
+"""Nautobot Circuit Maintenance plugin application level metrics exposed through nautobot_capacity_metrics."""
+from collections import OrderedDict
+import functools
+from datetime import datetime
+from prometheus_client.core import CounterMetricFamily
+from nautobot.circuits.models import CircuitTermination
+from django.conf import settings
+
+from .models import CircuitImpact, CircuitMaintenance
+
+
+def rgetattr(obj, attr, *args):
+    """Recursive GetAttr to look for nested attributes."""
+
+    def _getattr(obj, attr):
+        """Extract the nested value. If the value supports `all()` we return the first valid object."""
+        value = getattr(obj, attr, *args)
+
+        return value
+
+    return functools.reduce(_getattr, [obj] + attr.split("."))
+
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_circuit_maintenance", {})
+DEFAULT_LABELS = {
+    "circuit": "circuit.cid",
+    "provider": "circuit.provider.name",
+    "circuit_type": "circuit.type.name",
+    "site": "site.name",
+}
+
+
+def metric_circuit_operational():
+    """Expose the operational state of Circuits with a CircuitTermination when a Maintenance is ongoing.
+
+    # Circuit operational
+    circuit_maintenance_status{"circuit": "XXXXX", provider="ntt", circuit_type="peering", site='XX"} 1.0
+
+    # Circuit in maintenance mode
+    circuit_maintenance_status{"circuit": "YYYYYY", provider="ntt", circuit_type="peering", site='YY"} 2.0
```

Comment:
Some inconsistency here and in the docs - what's the actual metric name? Is it `circuit_maintenance_status` (here), `nautobot_circuit_maintenance` (below), or `nautobot_circuit_maintenance_total` (README.md)?